pennylane/interfaces/execution.py

```diff
@@ -363,53 +446,63 @@ def cost_fn(params, x):
             device_batch_transform=device_batch_transform,
         )
 
+    ### Specifying and preprocessing variables ####
+
     if interface == "auto":
         params = []
         for tape in tapes:
             params.extend(tape.get_parameters(trainable_only=False))
         interface = qml.math.get_interface(*params)
 
+    new_device_interface = isinstance(device, qml.devices.experimental.Device)
+    config = qml.devices.experimental.ExecutionConfig(interface=interface)
     gradient_kwargs = gradient_kwargs or {}
 
-    if device_batch_transform:
-        dev_batch_transform = set_shots(device, override_shots)(device.batch_transform)
-        tapes, batch_fn = qml.transforms.map_batch_transform(dev_batch_transform, tapes)
-    else:
-        batch_fn = lambda res: res  # pragma: no cover
-
     if isinstance(cache, bool) and cache:
         # cache=True: create a LRUCache object
         cache = LRUCache(maxsize=cachesize)
         setattr(cache, "_persistent_cache", False)
 
-    batch_execute = set_shots(device, override_shots)(device.batch_execute)
+    if new_device_interface:
+        batch_execute = device.execute
+    else:
+        batch_execute = set_shots(device, override_shots)(device.batch_execute)
 
-    if expand_fn == "device":
-        expand_fn = lambda tape: device.expand_fn(tape, max_expansion=max_expansion)
+    expand_fn = _preprocess_expand_fn(expand_fn, device, max_expansion)
 
-    if gradient_fn is None:
-        # don't unwrap if it's an interface device
-        if "passthru_interface" in device.capabilities() or device.short_name == "default.mixed":
-            return batch_fn(
-                qml.interfaces.cache_execute(
-                    batch_execute, cache, return_tuple=False, expand_fn=expand_fn
-                )(tapes)
-            )
-        unwrapped_tapes = tuple(qml.transforms.convert_to_numpy_parameters(t) for t in tapes)
-        res = qml.interfaces.cache_execute(
-            batch_execute, cache, return_tuple=False, expand_fn=expand_fn
-        )(unwrapped_tapes)
+    #### Executing the configured setup #####
 
-        return batch_fn(res)
+    tapes, batch_fn = _batch_transform(
+        tapes, device, config, override_shots, device_batch_transform
+    )
 
-    if gradient_fn == "backprop" or interface is None:
-        return batch_fn(
-            qml.interfaces.cache_execute(
-                batch_execute, cache, return_tuple=False, expand_fn=expand_fn
-            )(tapes)
+    # Exiting early if we do not need to deal with an interface boundary
+    no_interface_boundary_required = interface is None or gradient_fn in {None, "backprop"}
+    if no_interface_boundary_required:
+
+        device_supports_interface_data = (
+            new_device_interface
+            or config.interface is None
+            or gradient_fn == "backprop"
+            or device.short_name == "default.mixed"

```

Comment:
trying to remember why this patch is here, but maybe it's not relevant anymore with the new return types?

Reply:
default mixed also uses `qml.math` and works with all interfaces.